packages/swing-store/docs/data-export.md

```diff
@@ -179,22 +179,28 @@ As a result, for each active vat, the first-stage Export Data contains a record
 
-When importing, the `importSwingStore()` function takes an options bag, which has property named `includeHistorical`. This property defaults to `false`, which makes the importer ignore any historical artifacts present in the export dataset. To import the historical transcript spans (and snapshots), you must set it to `true`.
+However the `debug` export mode will never fail: it merely dumps everything in the swingstore, without limits or completeness checks.
 
-So, to convey historical transcript spans from one swingstore to another, you must set three options along the way:
+Note that `artifactMode` does not affect the Export Data generated by the exporter (because if we *ever* want to validate this optional data, the hashes are mandatory). It only affects the names returned by `getArtifactNames()`: the list will be smaller for `operational` than for `archival`. Re-exporting from a previously-pruned copy under `archival` mode will fail, because the second SwingStore cannot magically reconstruct the missing data.
+
+Also note that when a vat is terminated, we delete all information about it, including transcript items and snapshots, both current and old. This will remove all the Export Data records, and well as the matching artifacts from `getArtifactNames`.
+
+When importing, the `importSwingStore()` function's options bag takes a property named `artifactMode`, with the same meanings as for export. Importing with the `operational` mode will ignore any artifacts other than those needed for current operations, and will fail unless all such artifacts were available. Importing with `replay` will ignore spans from old incarnations, but will fail unless all spans from current incarnations are present. Importing with `archival` will fail unless all spans from all incarnations are present. There is no `debug` option during import.
+
+So, to convey current-incarnation historical transcript spans from one swingstore to another, you must set four options along the way:
 
 * the original swingstore must be opened with `{ includeHistorical: true }`, otherwise the old spans will be pruned immediately
-* the export must use `makeSwingStoreExporter(dirpath, 'archival')`, otherwise the export will omit the old spans
-* the import must use `importSwingStore(exporter, dirPath, { includeHistorical: true })`, otherwide teh import will ignore the old spans
+* the export must use `makeSwingStoreExporter(dirpath, { artifactMode: 'replay'})`, otherwise the export will omit the old spans
+* the import must use `importSwingStore(exporter, dirPath, { artifactMode: 'replay', includeHistorical: true })`, otherwise the import will ignore the old spans, and/or the newly created SwingStore will prune historical spans as new ones are created
```

Comment:
The story I was telling did not include being "restored": when I said "the original swingstore", I was talking about the one created by `initSwingStore`, before any export/import steps had happened.

You're right about `includeHistorical`, I was confusing it with the `makeSwingStore` option named `keepTranscripts`. I'll rewrite this section to say:

* the original swingstore must be opened with `{ keepTranscripts: true }` (which is the default), otherwise the old spans will be pruned immediately
* the export must use `makeSwingStoreExporter(dirpath, { artifactMode: 'replay'})`, otherwise the export will omit the old spans
* the import must use `importSwingStore(exporter, dirPath, { artifactMode: 'replay', keepTranscripts: true })` (i.e. `importSwingStore(exporter, dirPath, { artifactMode: 'replay' })` because keepTranscripts is the default), otherwise the import will ignore the old spans, and/or the newly created SwingStore will prune historical spans as new ones are created

(although given that `keepTranscripts` defaults to true, I wonder if I should just not even mention it)

(if `keepTranscripts: false`, then old spans are pruned immediately.. our vague idea was to expose this somehow in a config file, so validators who want to save space can flip the switch)

Reply:
Oh that `includeHistorical` was confusing. I actually wasn't aware of `keepTranscripts`. If we keep it in this explanation, let's describe what it does.

Reply:
Both `keepSnapshots` and `keepTranscripts` are part of the options bag, shared by `initSwingStore` and `openSwingStore`, and also by `importSwingStore` (because it returns a `SwingStore` instead of merely initializing one in-place.. have I mentioned that I'd be much happier if neither `initSwingStore` nor `importSwingStore` returned one?).

Both default to `false`: the assumption that swing-store/SwingSet makes is that you care about minimizing space, and only care about operational requirements, not anticipated exotic replay requirements.

cosmic-swingset is where our interests diverge from the swing-store/SwingSet default, and "we" (Agoric / the Agoric chain / `@agoric/cosmic-swingset` / `agd`) are willing to spend disk space (and export/import time) for the sake of this anticipated replay thing.

So I think it's ok for swing-store to have the simpler option name, with a default polarity that chooses for minimal space. It definitely is an optional boolean, but if you accept my argument that it's ok for swing-store to have that default, then I think it's optional in the correct direction.

Reply:
Ok now I understand why this is still confusing to me. `keepTranscript` has no effect on the import process, only during the Swingset execution when performing a `doSpanRollover`. Since we're NOT using the swing-store opened by `importSwingStore` for Swingset operations, I think we should just leave this out of this explanation.